integritybackend/actions.py

```diff
@@ -491,6 +493,95 @@ def c2pa_proofmode(self, zip_path: str, org_config: dict, collection_id: str):
             self._purge_from_tmp(tmp_img_dir, action_tmp_dir)
             self._purge_from_tmp(tmp_zip, action_tmp_dir)
 
+    def copy_proofmode(self, zip_path: str, org_config: dict, collection_id: str):
+        """Process a proofmode zip that bundles multiple assets with metadata,
+        and extracts them to the output folder.
+
+        Args:
+            zip_path: path to asset zip (will be copied, not altered)
+            org_config: configuration dictionary for this organization
+            collection_id: string with the unique collection identifier this
+                asset is in
+
+        Raises:
+            Exception if errors are encountered during processing
+        """
+        try:
+            # TODO: change function to take just org_id as param
+            action_name = "copy-proofmode"
+            org_id = org_config["id"]
+            asset_helper = AssetHelper(org_id)
+
+            # Get paths
+            action_dir = asset_helper.path_for_action(collection_id, action_name)
+            action_output_dir = asset_helper.path_for_action_output(
+                collection_id, action_name
+            )
+            action_tmp_dir = asset_helper.path_for_action_tmp(
+                collection_id, action_name
+            )
+
+            # Verify and copy zip
+            input_zip_sha = os.path.splitext(os.path.basename(zip_path))[0]
+            if input_zip_sha != _file_util.digest_sha256(zip_path):
+                raise Exception(f"SHA-256 of ZIP does not match file name: {zip_path}")
+            tmp_zip = shutil.copy2(zip_path, action_tmp_dir)
+            bundle_name = f"{input_zip_sha}-images"
+
+            # Define paths for files extracted from proofmode zip
+            tmp_img_dir = os.path.join(action_tmp_dir, bundle_name)
+            action_img_dir = os.path.join(action_dir, bundle_name)
+
+            meta_content = None
+            photographer_id = None
+            with ZipFile(tmp_zip) as zipf:
+                meta_content_path = next(
+                    (s for s in zipf.namelist() if s.endswith("-meta-content.json")),
+                    None,
+                )
+                if meta_content_path is None:
+                    raise Exception(f"ZIP at {zip_path} has no content metadata file")
+                with zipf.open(meta_content_path) as meta_content_f:
+                    meta_content = json.load(meta_content_f)["contentMetadata"]
+                    source_name = meta_content.get("private", {}).get("signal", {}).get("sourceName")
```

Comment:
```suggestion
                    source_name = meta_content.get("private", {}).get("author", {}).get("name")
```

CC: https://github.com/starlinglab/integrity-backend/pull/100/files#r873173729